src/taipy/core/common/_reload.py

```diff
@@ -10,29 +10,46 @@
 # specific language governing permissions and limitations under the License.
 
 import functools
+from typing import Dict
 
 
-@functools.lru_cache
-def _get_manager(manager: str):
-    from ..cycle._cycle_manager_factory import _CycleManagerFactory
-    from ..data._data_manager_factory import _DataManagerFactory
-    from ..job._job_manager_factory import _JobManagerFactory
-    from ..pipeline._pipeline_manager_factory import _PipelineManagerFactory
-    from ..scenario._scenario_manager_factory import _ScenarioManagerFactory
-    from ..task._task_manager_factory import _TaskManagerFactory
+class SingletonMeta(type):
```

Comment:
I'm not sure if this is a good place to put this class. It's only used here for now, so what do you think?

Reply:
I believe we don't even need the parent class. We can just merge both `SingletonMeta` and `Reloader`.